test/e2e/tests/subscription-uc.js

```diff
@@ -0,0 +1,106 @@
+
+function generateEmail() {
+    return 'keep.' + shortid.generate() + '@mailtrain.org';
+}
+
+async function subscribe(testUserEmail) {
+    const subscription = {
+        email: testUserEmail
+    };
+
+    await step('User navigates to list subscription page', async () => {
+        await page.webSubscribe.navigate();
+    });
+
+    await step('User submits a valid email', async () => {
+        await page.webSubscribe.enterEmail(testUserEmail);
+        await page.webSubscribe.submit();
+    });
+
+    await step('System shows a notice that further instructions are in the email', async () => {
+        await page.webConfirmSubscriptionNotice.waitUntilVisible();
+    });
+
+    await step('System sends an email with a link to confirm the subscription', async () => {
+        await page.mailConfirmSubscription.fetchMail(testUserEmail);
+    });
+
+    await step('User clicks confirm subscription in the email', async () => {
+        await page.mailConfirmSubscription.click('confirmLink');
+    });
+
+    await step('System shows a notice that subscription has been confirmed', async () => {
+        await page.webSubscribedNotice.waitUntilVisible();
+    });
+
+    await step('System sends an email with subscription confirmation', async () => {
+        await page.mailSubscriptionConfirmed.fetchMail(testUserEmail);
+        subscription.unsubscribeLink = await page.mailSubscriptionConfirmed.link('unsubscribeLink');
+        subscription.manageLink = await page.mailSubscriptionConfirmed.link('manageLink');
+    });
+
+    return subscription;
+}
+
+describe('Subscription use-cases', function() {
+    before(() => driver.manage().deleteAllCookies());
+
+    useCase('Subscription to a public list (main scenario)', async () => {
+        await subscribe(generateEmail());
+    });
+
+    useCase('Subscription to a public list (invalid email)', async () => {
+        await step('User navigates to list subscribe page', async () => {
+            await page.webSubscribe.navigate();
+        });
+
+        await step('User submits an invalid email', async () => {
+            await page.webSubscribe.enterEmail('foo@bar.nope');
+            await page.webSubscribe.submit();
+        });
+
+        await step('System shows a flash notice that email is invalid', async () => {
+            await page.webSubscribe.waitForFlash();
+            expect(await page.webSubscribe.getFlash()).to.contain('Invalid email address');
+        });
+    });
+
+    useCase('Unsubscription from list #1 (one-step, no form)', async () => {
+        const subscription = await subscribe(generateEmail());
+
+        await step('User clicks the unsubscribe button', async () => {
+            await page.mailSubscriptionConfirmed.click('unsubscribeLink');
+        });
+
+        await step('System show a notice that confirms unsubscription', async () => {
+            await page.webUnsubscribedNotice.waitUntilVisible();
```

Comment:
Roger, thanks a lot for doing this interim testing. Really greatly appreciated. I suspected this could be an issue.

Well, one "correct" solution that comes to my mind would be to extend waitUntilVisible with a piece of custom browser code something like document.mailTrainRefreshAcknowledged=true.

Then we could have a new function waitUntilVisibleAfterRefresh which would first wait for the document to lose the mailTraiRefreshAcknowledged property. Only then it would start waiting for the element.

Another options is to be able to wait for multiple items on the page and ensure that always at least one is distinguishing.

What do you think? I would possibly do both since they address slightly different cases.